libs/dyn/mapping.go

```diff
@@ -144,6 +137,7 @@ func (m Mapping) Clone() Mapping {
 // Merge merges the key-value pairs from another Mapping into the current Mapping.
 func (m *Mapping) Merge(n Mapping) {
 	for _, p := range n.pairs {
-		m.Set(p.Key, p.Value) //nolint:errcheck
+		// Following original Merge logic: do not take key locations from n
+		m.SetLoc(p.Key.MustString(), nil, p.Value)
```

Comment:
Not true; it takes locations for keys that are not present in `m`.

Reply:
thanks, passing location in 59244c2fe7123c24930a6e55db6f48fd48e9ae64